### PR TITLE
catch and log errors finding or spinning up sites rather than logging…

### PR DESCRIPTION
**Pull request by boutell**
… a lost promise rejection and leaking a socket

This is a possible resolution to a longstanding issue impacting our clients, and in any case it's a valuable fix!

I would like to see this approved by Anthony and Bea before publication, for their familiarity with it. I may test it as a branch for one client due to the urgency for them.

I tested this locally by inserting thrown exceptions before and after getting the site object, to verify it doesn't make matters worse and logs the errors effectively.

### CODE REVIEW

**Review by abea — APPROVED** [review]

**Comment by boutell** [comment]
Much better now; `npm test` completes and exits thanks to the new destroy mechanism, which allowed me to find bugs in apostrophe's destroy mechanism. NOTE: that won't happen unless you have the `destroy-clean-up-loaders` branch of `apostrophe` itself (or it has been merged).

Also, debugged and tested the catching of errors during the startup of an `apos` object. Significant bugs here were a likely source of resource leaks. To debug that, I temporarily threw an exception inside `modulesReady` in apostrophe when `initFailed` is present. The test suite fails, of course, but it reports the error in a useful way and exits properly, which means we're doing cleanup properly now. No more unhandled promise rejection.

**Review by abea — APPROVED** [review]

File: test/index.js [hunk 1]
```diff
@@ -151,5 +153,11 @@ describe('Apostrophe-multisite', function() {
       });
       expect(response).to.have.property('statusCode', 302);
     });
+
+    // it('set up why-is-node-running', function() {
+    //   setTimeout(function () {
+    //     running(); // logs out active handles that are keeping node running
+    //   }, 60000);
+    // });
```
Comment (falkodev) [inline test/index.js:161]: Need to remove these comments?

Comment (boutell) [reply test/index.js:161]: No, it is there to help with any future recurrence of issues with apos.destroy not behaving.

**Comment by boutell** [comment]
Merging as I answered Anthony's question explaining why the commented-out code has value to future debugging and there were no changes required.

**Pull request merged**